components/tools/OmeroPy/src/omero/rtypes.py

```diff
@@ -171,6 +169,8 @@ def rbool(val):
     """
     if val is None or isinstance(val, omero.RBool):
         return val
+    elif isinstance(val, basestring):
```

Comment:
I'm really torn on the impact of this specific conditional type coercion here. Having the semantics be different than `bool(val)` seems to me to be particularly dangerous and may lead to additional confusion.

At some level I would almost prefer us drawing a line in the sand with:

```
...
elif isinstance(val, basestring):
    raise TypeError('unsupported %s' % type(value))
...
```

Reply:
Having kicked off this discussion with this code snippet:

```
          boolValue=_field_info_data(wrapper=omero.rtypes.rbool, nullable=True),
```

I'll add I didn't intended to kick off any breaking changes. I'm happy for us to discuss what `rbool` _should_ have originally done and what it should do in the future, though this might not be the place.

For me, more immediately the question I was getting towards was:

> How can the cli (specifically, [obj.py](https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroPy/src/omero/plugins/obj.py)) work with string inputs?

i.e. we need some mechanism so that `wrapper` above can know it's dealing with string inputs and then do something smarter.

```
def rbool(val, parse_trueness=False):
```

is one quick idea though obviously it needs hammering on.